pkg/datapath/loader/base.go

```diff
@@ -107,6 +110,42 @@ func writePreFilterHeader(preFilter *prefilter.PreFilter, dir string) error {
 	return fw.Flush()
 }
 
+// Return the list of interfaces with egress masquerading.
+// - If option.Config.Masquerade is disabled, return an empty slice.
+// - If option.Config.EgressMasqueradeInterfaces is set to a simple interface
+//   name, return its value.
+// - If it is empty, return the list of all interfaces.
+// - If it has a '+' suffix, return the names of all matching interfaces.
+func getEgressMasqueradeInterfaces() ([]string, error) {
```

Comment:
> Style: The processing of interface names for the ip rule addition feels a bit cumbersome (with this + prefix to process). I'm not sure we have a helper for this somewhere, I could not find one (other lxc+-style names are passed directly to iptables from what I could see). We do have a helper somewhere for setting rp_filter but it is not used in Reinitialize() from the loader, so I just stuck to local code and appended to sysSettings.

I doubt we have another place where we actually process these. For most cases it has been sufficient to just pass the argument directly to `iptables`.

But this highlights a point, if we're going to have it here, seems reasonable to add a small unit test to check that the devices are being properly selected by this function. (Even if it's abstracted so we're not directly interacting with the kernel, just matching the string names properly).

Reply:
Right, a unit test is a good idea. I'll add one if we keep the relevant code, let's figure out how to get the interfaces on which we should load the rules first.

Reply:
Function is gone. Marking as resolved.